pkg/scheduler/scheduler.go

```diff
@@ -329,26 +343,28 @@ func (s *Scheduler) admit(ctx context.Context, e *entry) error {
 	return nil
 }
 
-// findFlavorForResources returns a flavor which can satisfy the resource request,
+// findFlavorForCodepResources returns a flavor which can satisfy the resource request,
 // given that wUsed is the usage of flavors by previous podsets.
 // If it finds a flavor, also returns any borrowing required.
-func findFlavorForResource(
+func findFlavorForCodepResources(
 	log logr.Logger,
-	name corev1.ResourceName,
-	val int64,
-	wUsed map[string]int64,
+	requests workload.Requests,
+	wUsed cache.ResourceQuantities,
 	resourceFlavors map[string]*kueue.ResourceFlavor,
 	cq *cache.ClusterQueue,
-	spec *corev1.PodSpec) (string, int64, *admissionStatus) {
+	spec *corev1.PodSpec) (string, map[corev1.ResourceName]int64, *admissionStatus) {
 	var status admissionStatus
 
-	if _, exists := cq.RequestableResources[name]; !exists {
-		return "", 0, asStatus(fmt.Errorf("resource unavailable in ClusterQueue"))
+	// Keep any resource name as an anchor to gather flavors for.
+	var rName corev1.ResourceName
+	for rName = range requests {
+		break
 	}
 
 	// We will only check against the flavors' labels for the resource.
-	selector := flavorSelector(spec, cq.LabelKeys[name])
-	for _, flvLimit := range cq.RequestableResources[name] {
+	// Since all the resources share the same flavors, they use the same selector.
+	selector := flavorSelector(spec, cq.LabelKeys[rName])
+	for i, flvLimit := range cq.RequestableResources[rName].Flavors {
```

Comment:
I guess we were not favoring flavors that don't cause borrowing, we just follow the flavor order in CQ; with this change, doing that will get a little more complicated, but I think still doable (like check all flavors, and return the one that causes the minimum borrowing across dependent resources).